### ISSUE TITLE
Sidekiq Pro: The gem was found in multiple relevant sources

### ISSUE THREAD

**Issue by mcfadden** [mperham/sidekiq#4815]
We use Sidekiq Pro. With nothing changing in our Gemfile, we started getting this error this morning when running `bundle install` on our development machines.

```ruby
Fetching gem metadata from https://gems.contribsys.com/..
Fetching gem metadata from https://rubygems.org/.........
Fetching gem metadata from https://gems.contribsys.com/.
Resolving dependencies....
The gem 'rack' was found in multiple relevant sources.
  * rubygems repository https://[removed]@gems.contribsys.com/ or installed locally
  * rubygems repository https://rubygems.org/ or installed locally
You must add this gem to the source block for the source you wish it to be installed from.
```

Our Gemfile looks something like:

```ruby
source 'https://rubygems.org'

source 'https://[removed]@gems.contribsys.com/' do
  gem 'sidekiq-pro'
end

...

gem 'rack'
```

We also saw the error with other gems including `connection_pool` and `concurrent-ruby`.

I traced it down to upgrading bundler to `2.2.10`. We worked around it for the moment by downgrading to `2.2.9`.

This appears to be directly related to [this security update](https://bundler.io/blog/2021/02/15/a-more-secure-bundler-we-fixed-our-source-priorities.html) in bundler 2.2.10

I don't expect you to provide support for bundler, but I assume a fair number of people will be impacted by this so I was hoping to get a suggested / preferred approach. Right now, it appears that it is trying to load the dependencies of `sidekiq-pro` from the contribsys repository, and that feels incorrect to me.

**Comment by mcfadden** [comment]
I think this is entirely a bundler bug: https://github.com/rubygems/rubygems/issues/4380

**Comment by connorshea** [comment]
FWIW Mike tweeted that the fix for now is to downgrade to bundler 2.2.9: https://twitter.com/getajobmike/status/1361372486604967940

I'd probably recommend doing that until the issue in bundler is resolved.

**Comment by jedrekdomanski** [comment]
I confirm that after installing bundler 2.2.10 I had a similar issue:
```
The gem 'redis' was found in multiple relevant sources.
  * rubygems repository https://enterprise.contribsys.com/ or installed locally
  * rubygems repository https://rubygems.org/ or installed locally
You must add this gem to the source block for the source you wish it to be installed from.
```
After downgrading to 2.1.4 the issue is gone.

**Comment by EugeneGrigorenko** [comment]
BTW, should be fixed now with 2.2.11 release

**Comment by mcfadden** [comment]
I also confirmed it's fixed with bundler 2.2.11
